source/light_utils.h

```diff
@@ -1,8 +1,17 @@
 #include "board.h"
 #include "hal.h"
 
+/*
+    Structs
+*/
+// Struct defining an LED and its RGB color components
+typedef struct led{
+    uint8_t red, green, blue;
+}led;
```

Comment:
Please name struct with _t. Also you don't have to use named struct if you are not referencing the struct within itself.

```C
typedef struct {
    uint8_t red;
    uint8_t green;
    uint8_t blue;
} led_t;
```